test/python/transpiler/test_normalize_rx_angle.py

```diff
@@ -0,0 +1,147 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the NormalizeRXAngle pass"""
+
+import unittest
+import numpy as np
+from ddt import ddt, named_data
+
+from qiskit import QuantumCircuit
+
+from qiskit.transpiler.passes.optimization.normalize_rx_angle import (
+    NormalizeRXAngle,
+)
+from qiskit.test import QiskitTestCase
+from qiskit.providers.fake_provider import FakeBelemV2
+from qiskit.transpiler import Target
+
+
+@ddt
+class TestNormalizeRXAngle(QiskitTestCase):
+    """Tests the NormalizeRXAngle pass."""
+
+    @staticmethod
+    def count_gate_number(gate, circuit):
+        """Count the number of a specific gate type in a circuit"""
+        if gate not in QuantumCircuit.count_ops(circuit):
+            gate_number = 0
+        else:
+            gate_number = QuantumCircuit.count_ops(circuit)[gate]
+        return gate_number
+
+    def test_not_convert_to_X_if_no_calib_in_target(self):
+        """Check that RX(pi) is NOT converted to X,
+        if X calibration is not present in the target"""
+        empty_target = Target()
+        tp = NormalizeRXAngle(target=empty_target)
+
+        qc = QuantumCircuit(1)
+        qc.rx(90, 0)
+
+        transpiled_circ = tp(qc)
+        self.assertEqual(self.count_gate_number("x", transpiled_circ), 0)
+
+    def test_SX_conversion_works(self):
+        """Check that RX(pi/2) is converted to SX,
+        if SX calibration is present in the target"""
+        backend = FakeBelemV2()
+        tp = NormalizeRXAngle(target=backend.target)
```

Comment:
```suggestion
        target = Target()
        target.add_instruction(SXGate())
        tp = NormalizeRXAngle(target=target)
```
This is more explicit?

Reply:
I agree. Do you think I should apply this approach to other tests as well and avoid importing `FakeBelemV2()` altogether? Or do you think that the tests would become more robust if I utilize the predefined instance of a backend?

Reply:
Weirdly, this test passes only when I do 
``` python
target.add_instruction(SXGate(), properties={(0,): None})
```
And fails if I do
``` python
target.add_instruction(SXGate(), properties={None: None})
```
Even if the manual says that setting `None` as the only key of the `properties` dictionary means that the instruction is available to all qubits. https://qiskit.org/documentation/stable/0.35/stubs/qiskit.transpiler.Target.add_instruction.html

Reply:
I often avoid using fake backend for unittest because the input object is too complicated. But this object is good for writing an integration test. What kind of error did you have?